### ISSUE TITLE
Fix dependency warnings when building gem

### ISSUE THREAD

**Issue by kyletolle** [fulcrumapp/fulcrum-ruby#6]
Currently, when I build the gem, I get these warnings:

```
WARNING:  open-ended dependency on rake (>= 0, development) is not recommended
  if rake is semantically versioned, use:
    add_development_dependency 'rake', '~> 0'
WARNING:  open-ended dependency on rspec (>= 0, development) is not recommended
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 0'
WARNING:  open-ended dependency on webmock (>= 0, development) is not recommended
  if webmock is semantically versioned, use:
    add_development_dependency 'webmock', '~> 0'
WARNING:  open-ended dependency on activesupport (>= 0) is not recommended
  if activesupport is semantically versioned, use:
    add_runtime_dependency 'activesupport', '~> 0'
WARNING:  open-ended dependency on faraday_middleware (>= 0) is not recommended
  if faraday_middleware is semantically versioned, use:
    add_runtime_dependency 'faraday_middleware', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
```

I suppose there's something to be said for leaving it open for people to use RSpec 2, and ActiveSupport 3, but should we add versions for these gems that are known to work?

People using or developing on the gem will automatically get newer versions of the gems, which might actually cause issues.

**Comment by zhm** [comment]
Fixed by #7!

**Comment by snkashis** [comment]
Is the move to ActiveSupport 4.2 actually required for something specific? I feel like a lot of people(including us) are still on 4.1.
